packages/venia-ui/lib/components/CategoryTree/categoryTree.js

```diff
@@ -30,10 +30,15 @@ const Tree = props => {
     // for each child category, render a direct link if it has no children
     // otherwise render a branch
     const branches = Array.from(childCategories, childCategory => {
-        const [id, { category, isLeaf }] = childCategory;
+        const [id, { category, isLeaf, parentCategory }] = childCategory;
 
         return isLeaf ? (
-            <Leaf key={id} category={category} onNavigate={onNavigate} />
+            <Fragment key={id}>
+                {parentCategory && (
+                    <Leaf category={parentCategory} onNavigate={onNavigate} />
```

Comment:
Once the `childCategory` includes information on whether this is the parent category, it can pass that to the `Leaf` component to decide how to differentiate it in the display (ex: add the "All" text so "Bottoms" becomes "All Bottoms").